contracts/kernel/Kernel.sol

```diff
@@ -5,227 +5,52 @@ import "./KernelStorage.sol";
-            _manager
-        );
+        setApp(APP_BASES_NAMESPACE, ACL_APP_ID, baseACL);
+        setApp(APP_ADDR_NAMESPACE, ACL_APP_ID, _acl);
+        _acl.initialize(_permissionsCreator);
     }
 
-    /**
-    * @dev Grants a permission if allowed. This requires `msg.sender` to be the permission manager
-    * @notice Grants `_entity` the ability to perform actions of role `_role` on `_app`
-    * @param _entity Address of the whitelisted entity that will be able to perform the role
-    * @param _app Address of the app in which the role will be allowed (requires app to depend on kernel for ACL)
-    * @param _role Identifier for the group of actions in app given access to perform
-    */
-    function grantPermission(address _entity, address _app, bytes32 _role)
-        onlyPermissionManager(_app, _role)
-        external
-    {
-        require(!hasPermission(_entity, _app, _role));
-        _setPermission(
-            _entity,
-            _app,
-            _role,
-            true
-        );
-    }
-
-    /**
-    * @dev Revokes permission if allowed. This requires `msg.sender` to be the parent of the permission
-    * @notice Revokes `_entity` the ability to perform actions of role `_role` on `_app`
-    * @param _entity Address of the whitelisted entity that will be revoked access
-    * @param _app Address of the app in which the role is revoked
-    * @param _role Identifier for the group of actions in app given access to perform
-    */
-    function revokePermission(address _entity, address _app, bytes32 _role)
-        onlyPermissionManager(_app, _role)
-        external
-    {
-        require(hasPermission(_entity, _app, _role));
-        _setPermission(
-            _entity,
-            _app,
-            _role,
-            false
-        );
-    }
-
-    /**
-    * @notice Sets `_newManager` as the manager of the permission `_role` in `_app`
-    * @param _newManager Address for the new manager
-    * @param _app Address of the app in which the permission management is being transferred
-    * @param _role Identifier for the group of actions in app given access to perform
-    */
-    function setPermissionManager(address _newManager, address _app, bytes32 _role)
-        onlyPermissionManager(_app, _role)
-        external
-    {
-        _setPermissionManager(_newManager, _app, _role);
-    }
-
-    /**
-    * @dev Changes appCode reference for `_appId`. This role is required before an app with a certain appId working properly
-    * @notice Upgrade app code of `_appId` to new implementation at address `_code` (CRITICAL!)
-    * @param _appId Namehash of the app name
-    * @param _code Address of new implementation for app
-    */
-    function setAppCode(bytes32 _appId, address _code) auth(UPGRADE_APPS_ROLE) external {
-        appCode[_appId] = _code;
-        SetAppCode(_appId, _code);
+    function setApp(bytes32 namespace, bytes32 name, address app) auth(APP_MANAGER) public returns (bytes32 id) {
```

Comment:
I think `setApp()` is trying to do do many things at once.

From what I gather, it:
- Locates the kernel implementation
- Locates base implementations of apps
- Locates "special" installed apps like the EVMScriptRegistry or ACL

Maybe it would make sense to put the last one in another mapping or give it a separate API?

(I also like `setAppCode` for clarity)